conda/core/index.py

```diff
@@ -145,12 +145,12 @@ def _supplement_index_with_cache(index):
             index[pcrec] = pcrec
 
 
-def _make_virtual_package(name, version=None, build_string='0'):
+def _make_virtual_package(name, version=None, build_string=None):
     return PackageRecord(
             package_type=PackageType.VIRTUAL_SYSTEM,
             name=name,
             version=version or '0',
-            build_string=build_string,
+            build_string=build_string or '0',
```

Comment:
Is it important to have an empty build string

Reply:
FWIW this was the previous behavior: https://github.com/conda/conda/blob/22.9.0/conda/core/index.py#L152-L163

Reply:
yes let's not change stuff because it is not obvious that we need it. too much history in this code base.

Reply:
We'd probably break stuff with empty strings. There is code similar to this in `resolve` for example:
```py
prec = PackageRecord(name="a", version="1", build="", build_number=0)
spec = MatchSpec(f"{prec.name} {prec.version} {prec.build}")
# this results in:
str(spec) == "a==1"
spec.get("build") is None
# i.e., with that MatchSpec from string above, we loose the build string completely
```